components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesBookmarksStorage.kt

```diff
@@ -178,11 +182,63 @@ open class PlacesBookmarksStorage(context: Context) : PlacesStorage(context), Bo
     override suspend fun sync(authInfo: AuthInfo): SyncStatus {
         return try {
             withContext(scope.coroutineContext) {
-                places.syncBookmarks(authInfo.into())
+                val ping = places.syncBookmarks(authInfo.into())
+                submitBookmarksPing(ping)
                 SyncStatus.Ok
             }
         } catch (e: PlacesException) {
             SyncStatus.Error(e)
         }
     }
 }
+
+@Suppress("ComplexMethod", "NestedBlockDepth")
+private fun submitBookmarksPing(ping: SyncTelemetryPing) {
+    ping.syncs.forEach { sync ->
+        sync.engines.forEach eachEngine@{ engine ->
+            if (engine.name != "bookmarks") {
+                return@eachEngine
+            }
+            BookmarksSync.apply {
+                val base = BaseGleanSyncPing.fromEngineInfo(ping.uid, engine)
+                uid.set(base.uid)
+                startedAt.set(base.startedAt)
+                finishedAt.set(base.finishedAt)
+                if (base.applied > 0) {
```

Comment:
nit: I'd drop a comment here specifying that, since all these metrics have `ping` lifetime, they get cleared after the ping is generated, so it's ok to simply add to them.